suite-native/device/src/middlewares/deviceMiddleware.ts

```diff
@@ -63,10 +57,14 @@ export const prepareDeviceMiddleware = createMiddlewareWithExtraDeps(
             dispatch(handleDeviceDisconnect(action.payload));
         }
 
+        const isConnectEnabled = selectIsDeviceConnectFeatureFlagEnabled(getState());
+
         switch (action.type) {
             case DEVICE.CONNECT:
             case DEVICE.CONNECT_UNACQUIRED:
-                dispatch(handleDeviceConnect(action.payload));
+                if (isConnectEnabled) {
```

Comment:
Why is this `if` necessary? Should not it happen always? IMO the fake device will never trigger `DEVICE.CONNECT`, so it will be ignored anyway in cases where is `selectIsDeviceConnectFeatureFlagEnabled ` false.

Reply:
Mostly as a safety net. I'd rather have it here than not.